### PR TITLE
test: add swap e2e tests on Tenderly network

### PR DESCRIPTION
**Pull request by davibroc**
## **Description**

These tests cover the following functionality:
- Token Detection
- Adding Network Manually (Tenderly)
- Fetching a Quote
- Swapping Token
- Swapping Token after Switching Network
- Activity List Verification

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25060?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. yarn playwright install chromium
2. yarn test:e2e:swap

## **Screenshots/Recordings**

https://github.com/MetaMask/metamask-extension/assets/6082863/5b676fb5-7416-42f6-8e86-5e47a6bf68a7

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -65,11 +65,10 @@ test-results/
 
 # MMI Playwright
 public/playwright
-test/e2e/mmi/.env
-test/e2e/mmi/playwright.config.ts
-test/e2e/mmi/specs/**/*-darwin.png
-test/e2e/mmi/dist/
-
+test/e2e/playwright/.env
```
Comment (zone-live) [inline .gitignore:68]: shouldn't it be `test/e2e/playwright/mmi/.env`?

**Review by zone-live — COMMENTED** [review]
@davibroc it looks good, just left one question.

File: .gitignore (hunk 1 above)
Comment (davibroc) [reply .gitignore]: Yes that's correct. I have fixed it

**Review by zone-live — DISMISSED** [review]

**Comment by dan437** [comment]
Great job adding Playwright tests for Swaps, they seem fast. Also nice to see that we are using real API calls with Playwright.

File: playwright/swap/pageObjects/activity-list-page.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    this.page = page;
+    this.activityItem = this.page
+      .getByTestId('activity-list-item-action')
+      .first();
+
+    this.status = this.page.locator('.transaction-status-label').first();
+  }
+
+  async checkActivityIsConfirmed(options: any) {
+    // await this.page.waitForTimeout(20000000);
```
[the remark this answers is not in the document]
Comment (davibroc) [reply playwright/swap/pageObjects/activity-list-page.ts:20]: Done

File: playwright/swap/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+1. Installing Playwright
+
+If you run the tests for first time, you may need to install the browser dependency (Playwright will inform you in case you need):
+
+```
+yarn playwright install chromium
+```
+
+2. Prepare the Build:
```
[the remark this answers is not in the document]
Comment (davibroc) [reply playwright/swap/README.md:15]: Done

File: ui/components/app/add-network/add-network.js [hunk 4]
```diff
@@ -256,6 +256,7 @@ const AddNetwork = () => {
                   <Button
                     type="inline"
                     className="add-network__add-button"
+                    data-testid={`add-network__${item.nickname}`}
```
[the remark this answers is not in the document]
Comment (davibroc) [reply ui/components/app/add-network/add-network.js:259]: Done

File: playwright/swap/pageObjects/extension-page.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import path from 'path';
+import { type Page, chromium } from '@playwright/test';
+
+const wait = (n) => new Promise((resolve) => setTimeout(resolve, n));
+
+const extensionPath = path.join(__dirname, '../../../../../dist/chrome');
+
+export class ChromeExtensionPage {
```
[the remark this answers is not in the document]
Comment (davibroc) [reply playwright/swap/pageObjects/extension-page.ts:8]: done

File: playwright/swap/pageObjects/activity-list-page.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { expect, type Locator, type Page } from '@playwright/test';
+
+export class ActivityListPage {
```
[the remark this answers is not in the document]
Comment (davibroc) [reply playwright/swap/pageObjects/activity-list-page.ts:3]: done

File: playwright/swap/pageObjects/network-controller-page.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    await this.approveBtn.click();
+    await this.switchToNetworkBtn.click();
+    await this.gotItBtn.click();
+  }
+
+  async selectNetwork(networkName) {
+    await this.networkDisplay.click();
+    await this.networkSearch.fill(networkName);
+    await this.page.getByText(networkName).click();
+    await this.page.waitForTimeout(2000);
```
[the remark this answers is not in the document]
Comment (davibroc) [reply playwright/swap/pageObjects/network-controller-page.ts]: I have removed all the waitForTimeout

**Review by dan437 — DISMISSED** [review]
Looks great, thanks for adding more Playwright tests, which I believe is the future for the extension 🎉

**Review by zone-live — DISMISSED** [review]

File: .circleci/config.yml [hunk 8]
```diff
@@ -818,6 +824,31 @@ jobs:
           path: builds-test-mmi-playwright
           destination: builds-test-mmi-playwright
 
+  prep-build-test-swap-playwright:
```
Comment (Gudahtt) [inline .circleci/config.yml:827]: Could you elaborate on why we need a new build here? We have been spending way too much CircleCI credits recently and have been trying to cut costs in any way we can, such as by removing builds and workflows. But we are adding one here.

Comment (Gudahtt) [reply .circleci/config.yml:827]: For example, could this new test use the pre-existing production or test build? I would expect playwright tests to generally use the test build, though this one seems to use one equivalent to the production build.

Comment (davibroc) [reply .circleci/config.yml]: @Gudahtt @dan437 I have found a pre-existing production build that works so we don't have to generate a new one. Please review again

File: .circleci/config.yml [hunk 9]
```diff
@@ -1386,9 +1389,9 @@ jobs:
       - run:
           name: test:e2e:chrome:mmi
           command: |
-            TESTFILES=$(circleci tests glob "test/e2e/mmi/**/*.spec.ts")
+            TESTFILES=$(circleci tests glob "test/e2e/playwright/mmi/**/*.spec.ts" | circleci tests split)
```
Comment (Gudahtt) [inline .circleci/config.yml:1392]: Hmm, why was `circleci tests split` added here? The docs for it seem to suggest that it's not recommended anymore:

https://circleci.com/docs/parallelism-faster-jobs/#the-tests-split-command



>The circleci tests split command has been superceded by circleci tests run as this command also gives you access to the [rerun failed tests](https://circleci.com/docs/rerun-failed-tests/) options.

**Review by Gudahtt — COMMENTED** [review]
Had one question about the CircleCI config, but overall LGTM! I'll leave the test changes to the other reviewers.

File: .circleci/config.yml (hunk 9 above)
Comment (davibroc) [reply .circleci/config.yml]: Good catch. It was not needed so I removed it

**Review by dan437 — APPROVED** [review]

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**